keras/layers/core.py

```diff
@@ -33,6 +33,11 @@ class Layer(object):
             (e.g. `(32, 100)` for a batch of 32 100-dimensional inputs).
     '''
     def __init__(self, **kwargs):
+        if not hasattr(self, 'params'):
```

Comment:
This change is necessary because we need to ensure that `non_trainable_weights` exists before applying the `kwargs`. Otherwise, some layers internally call `set_weights`, which in turn fails because `non_trainable_weights` doesn't exist. It only seemed natural to also move the check for `params` up to the same place.